paddle/operators/sequence_pool_op.h

```diff
@@ -135,6 +139,19 @@ class SequencePoolGradKernel : public framework::OpKernel<T> {
           in_g_e.device(place) =
               (out_g_e / std::sqrt(static_cast<T>(h))).broadcast(bcast);
           break;
+        case MAX: {
+          auto in_t = in->Slice<T>(static_cast<int>(lod[i]),
+                                   static_cast<int>(lod[i + 1]));
+          auto out_t = out->Slice<T>(i, i + 1);
+          auto in_e = EigenMatrix<T>::From(in_t, {h, w});
+          auto out_e = EigenMatrix<T>::From(out_t, {1, w});
+          auto equals = in_e == out_e.broadcast(bcast);
+          auto ones = in_g_e.constant(1);
+          auto zeros = in_g_e.constant(0);
+          in_g_e.device(place) =
+              out_g_e.broadcast(bcast) * equals.select(ones, zeros);
```

Comment:
It's better to keep the same way with the MaxSeqPool in the old framework, which records the max index in the forward and uses the max index in the backward. Since there may be multiple minimum values.

Reply:
Done. 在反向中采用maxCoeff函数，获得每列中的第一个最大值。
考虑到inference的效率，保持前向不变，maxCoeff加在反向中。
目前实现的版本性能上还有待优化，等之后的PR修复。